server.js

```diff
@@ -0,0 +1,386 @@
+          if (error) {
+            console.log(request.url + " cannot be found");
+          } else {
+            response.writeHead(200, { "Content-type": "text/css" });
+            response.write(data);
+            response.end();
+          }
+        });
+        // GET (request.url)
+      } else if (request.url) {
+        fs.readFile("./public" + request.url, function(error, data) {
+          if (error) {
+            console.log(request.url + " cannot be found");
+          } else {
+            response.writeHead(200, { "Content-type": "text/html" });
+            response.write(data);
+            response.end();
+          }
+        });
+      }
+    }
+  }
+
+  // POST REQUESTS
+
+  if (request.method === "POST") {
+    request.on("data", function(data) {
+      // Parse the POST data
+      let formDataObject = {};
+      let formDataArray = data.toString().split("&");
+
+      for (let i in formDataArray) {
+        let keys = formDataArray[i].split("=");
+        formDataObject[keys[0]] = keys[1];
+      }
+      // Return 500 if request is incomplete
+      if (
+        !formDataObject["elementName"] ||
+        !formDataObject["elementSymbol"] ||
+        !formDataObject["elementAtomicNumber"] ||
+        !formDataObject["elementDescription"]
+      ) {
+        console.log("Request is missing one or more values");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Requeste is missing one or more values"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Return 500 if file name is index or 404
+      } else if (
+        formDataObject["elementName"] === "index" ||
+        formDataObject["elementName"] === "404"
+      ) {
+        console.log("Invalid elementName");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Invalid elementName"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Create new HTML file and populate with POST data
+        let newElement = `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>The Elements - ${formDataObject["elementName"]}</title>
+    <link rel="stylesheet" href="/css/styles.css" />
+  </head>
+  <body>
+    <h1>${formDataObject["elementName"]}</h1>
+    <h2>${formDataObject["elementSymbol"]}</h2>
+    <h3>Atomic number ${formDataObject["elementAtomicNumber"]}</h3>
+    <p>
+      ${formDataObject["elementDescription"]}
+    </p>
+    <p><a href="/">back</a></p>
+  </body>
+</html>
+`;
+        fs.writeFile(
+          "./public/" + formDataObject["elementName"].toLowerCase() + ".html",
+          newElement,
+          function(error) {
+            if (error) throw error;
```

Comment:
Errors should be handled properly, they should send back a response with a appropriate status code and message.